data/scripts/scenarios.py

```diff
@@ -101,99 +104,99 @@ def from_ms(time):
+                length_hospital_stay = default["lengthHospitalStay"],
+                length_icu_stay = default["lengthICUStay"],
+                overflow_severity = default["overflowSeverity"],
+                peak_month = default["peakMonth"],
+                r0 = float(max(1, round(FIT_CASE_DATA[region]['r0'], 1)) if region in FIT_CASE_DATA else default["r0"]),
+                seasonal_forcing = default["seasonalForcing"])
+
+class ContainmentParams(schema.ContainmentData):
+    def __init__(self):
+        default = DEFAULTS["ContainmentData"]
+        return super(ContainmentParams, self).__init__([], default["numberPoints"])
 
-class DateRange(Object):
-    def __init__(self, tMin, tMax):
-        self.tMin = tMin
-        self.tMax = tMax
-
-class SimulationParams(Object):
+class SimulationParams(schema.SimulationData):
     def __init__(self, region):
-        tMin = FIT_CASE_DATA[region]['tMin'] if region in FIT_CASE_DATA else "2020-03-01"
-        tMax = "2020-09-01"
-        self.simulationTimeRange  = DateRange(tMin, tMax)
-        self.numberStochasticRuns = 0
+        return super(SimulationParams, self).__init__( \
+                simulation_time_range = DateRange( \
+                    datetime.strptime(FIT_CASE_DATA[region]['tMin'] if region in FIT_CASE_DATA else "2020-03-01", '%Y-%m-%d'),
+                    datetime.strptime("2020-09-01", '%Y-%m-%d')),
+                number_stochastic_runs = 0.0)
 
 # TODO: Region and country provide redudant information
 #       Condense the information into one field.
-class AllParams(Object):
+class AllParams(schema.AllParams):
     def __init__(self, region, country, population, beds, icus, hemisphere, srcPopulation, srcHospitalBeds, srcICUBeds):
-        self.population      = PopulationParams(region, country, population, beds, icus)
-        self.epidemiological = EpidemiologicalParams(region, hemisphere)
-        self.simulation      = SimulationParams(region)
-        self.containment     = ContainmentParams()
-        # uncomment if the sources are needed in the app
-        #self.sources         = {'populationServed': srcPopulation, 'hospitalBeds': srcHospitalBeds, 'ICUBeds': srcICUBeds }
+        #self.sources  = {'populationServed': srcPopulation, 'hospitalBeds': srcHospitalBeds, 'ICUBeds': srcICUBeds }
+        return super(AllParams, self).__init__( \
+                ContainmentParams(),
+                EpidemiologicalParams(region, hemisphere),
+                PopulationParams(region, country, population, beds, icus),
+                SimulationParams(region)
+        )
 
 # ------------------------------------------------------------------------
 # Functions
 
-def marshalJSON(obj, wtr):
+def marshalJSON(obj, wtr=None):
     """ Validate and store data to .json file
     Arguments:
     - obj: a dict of allParams
     """
+    if wtr is None:
+        return json.dumps(self, default=lambda x: x.__dict__, sort_keys=True, indent=4)
```

Comment:
the argument `self` should probably be `obj`.

Reply:
You are correct. It's fixed in the latest commit.